Refactor smart comment parsing (#1908)
graphile/crystal: postgraphile/website/postgraphile/smart-comments.md

$$;
```

> Note that adding two newlines in a row will be interpreted as the start of a multi-line comment, no tags added after the first
> newline will be parsed & will be treated as part of the description.

Reviewer: ```suggestion
Note that adding two newlines in a row separates the smart tags section from the
descriptive prose. Content after two newlines will not be parsed for smart tags
even if it starts with an `@` character.
```

$$;
```

Note that adding two newlines in a row separates the smart tags section from the
descriptive prose. Content after two newlines will not be parsed for smart tags

> even if it starts with an `@` character.
> 

Reviewer: ```suggestion
even if it starts with an `@` character.

For example:

```

```sql
comment on column my_schema.my_table.my_column is $$
@name meta
@isImportant
@jsonField date timestamp
@jsonField name text

This field has a load of arbitrary tags.
@jsonField episode enum ONE=1 TWO=2
$$;
```

> Results in the following JSON tags object:

Reviewer: ```suggestion
results in the following JSON tags object:
```
